### PR TITLE
Integrate testing on Windows using Azure Pipelines

### PR DESCRIPTION
**Pull request by desilinguist**
Addresses #308. 

- Add `azure-pipelines.yml` file to configure the build system on Azure.
- Add` DistributeTests.ps1` powershell script to divide up tasks on 5 parallel agents just like Travis CI.

### CODE REVIEW

**Comment by desilinguist** [comment]
Weird, I wonder why the Azure build shows as passing even though one of the sub-builds fails.

**Comment by desilinguist** [comment]
Okay, I changed the the config file to not do that.

**Comment by desilinguist** [comment]
There's a [single test failing](https://dev.azure.com/EducationalTestingService/RSMTool/_testManagement/runs?runId=158&_a=resultSummary&resultId=100081) on Windows. And it's failing because `numpy.int` maps to a C long which basically maps to a 32-bit number on Windows instead of 64-bit.

**Comment by desilinguist** [comment]
Okay, I have fixed the test and this PR is ready for review @aloukina @jbiggsets.

File: tests/test_reader.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def test_container_train_property(self):
                       [('train', 'csv'), ('test', 'xls'), ('train_metadata', 'tsv')],
                       [('train', 'jsonlines'), ('test', 'jsonlines')]]
 
-        converter = {'id': str, 'feature1': int, 'feature2': int, 'candidate': str}
+        converter = {'id': str, 'feature1': np.int64, 'feature2': np.int64, 'candidate': str}
```
Comment (jbiggsets) [inline tests/test_reader.py]: Just curious why these needed to be changed.

File: DistributeTests.ps1 [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    $testsToRun = $testsToRun + "tests/test_writer.py"
+    $testsToRun = $testsToRun + "tests/test_experiment_rsmtool_3.py"
+}
+elseif ($agentNumber -eq 5) {
+    $testsToRun = $testsToRun + "tests/test_experiment_rsmpredict.py"
+    $testsToRun = $testsToRun + "tests/test_reader.py"
+    $testsToRun = $testsToRun + "tests/test_reporter.py"
+    $testsToRun = $testsToRun + "tests/test_transformer.py"
+    $testsToRun = $testsToRun + "tests/test_utils.py"
+    $testsToRun = $testsToRun + "tests/test_experiment_rsmtool_4.py"
```
[the remark this answers is not in the document]
Comment (desilinguist) [reply DistributeTests.ps1]: Yeah, let’s do that in a separate PR for both Travis and Azure.

File: tests/test_reader.py (hunk 1 above)
Comment (desilinguist) [reply tests/test_reader.py]: See comment above.

File: DistributeTests.ps1 [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+# join all test files seperated by space. pytest runs multiple test files in following format pytest test1.py test2.py test3.py
+$testFiles = $testsToRun -Join " "
+Write-Host "Test files $testFiles"
+# write these files into variable so that we can run them using pytest in subsequent task.
```
Comment (aloukina) [inline DistributeTests.ps1]: Is there a way we can add a check to make sure all new test_* files are added?

Comment (desilinguist) [reply DistributeTests.ps1]: I can't really think of any obvious way to do the right now.

**Pull request merged**